### DISCUSSION TITLE
pac commands not working properly after recent version upgrade of the pac tool

### DISCUSSION THREAD

**Discussion by gituseranh** [microsoft/powerplatform-build-tools#801] [Q&A]
Hi! 
I recently upgraded pac to version 1.31.6 (Latest) on two systems. It is working fine on one but not on the other!  Not sure how the discrepancy happened! Apparently, I need to force the system to use **pac.cmd** instead of pac.exe but I do not know how to do that - would appreciate your suggestions or help with this issue.  Thank you in advance. [More details provided below]  

[On the system where it Does Not Work properly]
> Get-Command pac | Format-List
Name            : pac.exe
CommandType     : Application
Definition      : F:\build_apps\powerapps-cli\Microsoft.PowerApps.CLI.1.31.6\tools\pac.exe
Extension       : .exe
Path            : F:\build_apps\powerapps-cli\Microsoft.PowerApps.CLI.1.31.6\tools\pac.exe
FileVersionInfo : File: F:\build_apps\powerapps-cli\Microsoft.PowerApps.CLI.1.31.6\tools\pac.exe

                  InternalName:     pac.exe
                  OriginalFilename: pac.exe
                  FileVersion:      1.31.6
                  FileDescription:  Microsoft Power Platform CLI
                  Product:          Microsoft Power Platform©
                  ProductVersion:   1.31.6+g9147a23
----------------------------------------------------------------------
[On the system where it Works]
> Get-Command pac | Format-List
Name            : pac.cmd
CommandType     : Application
Definition      : F:\build_apps\powerapps-cli\pac.cmd
Extension       : .cmd
Path            : F:\build_apps\powerapps-cli\pac.cmd
FileVersionInfo : File:             F:\build_apps\powerapps-cli\pac.cmd
                  InternalName:
                  OriginalFilename:
                  FileVersion:
                  FileDescription:
                  Product:
                  ProductVersion:

**Comment by petrochuk** [comment]
there are two different flavors of pac cli.

1. Compiled with .Net core and is cross platform compatible (installed with .net tool)
2. Compiled with full .Net Framework which is only runs on Windows (installed with MSI installer) and usually invoked with pac.cmd which starts pac.launcher which checks for updates and then starts pac.exe for Windows only.

I suspect you use different flavors in these cases

**Comment by gituseranh** [comment]
I had an older version of the pac tool already installed on the systems [identical] and I just ran **pac install latest** ... the working version was upgraded from a 'more recent' older version [1.29.11] and the other from 1.21.13 which had already become outdated! Hence the discrepancy [I think] - but now, if there was a way I could make pac use the **pac.cmd** instead of pac.exe - probably that would make the latest version work properly on the other system as well.

**Comment by petrochuk** [comment]
pac.cmd is just a text file with two lines in it:

```
@echo off
"%~dp0pac.launcher.exe" %*
```

if you want to you can create it anywhere on your system and add to %PATH%

**Comment by gituseranh** [comment]
Ah!  I wasn't thinking :-) thank you so much - it worked!!!
